### PR TITLE
Fix SpotifyId base 62 and 16 str decoding

### PR DESCRIPTION
**Pull request by lelloman**
A SpotifyId is expected to be a 128 bits integer and can be parsed from a base 62 or 16 string. However the parsing functions only checked the validity of the characters of the string, but not its length. This could result in integer overflows or the parsing of incorrect strings as Spotify ids.

This commit add some checks to the length of the input string passed to the parse functions, and also checks for integer overflows in case of base62 encoded strings.

### CODE REVIEW

File: core/src/spotify_id.rs [hunk 1]
```diff
@@ -102,6 +102,9 @@ impl SpotifyId {
     ///
     /// [Spotify ID]: https://developer.spotify.com/documentation/web-api/concepts/spotify-uris-ids
     pub fn from_base16(src: &str) -> SpotifyIdResult {
+        if src.len() > 32 {
```
Comment (roderickvd) [inline core/src/spotify_id.rs:105]: Should this check be ```!= 32```?

**Review by roderickvd — COMMENTED** [review]
Thanks! One small question.

**Review by roderickvd — CHANGES_REQUESTED** [review]
Please check the comments on line 105.

**Comment by lelloman** [comment]
I guess that https://github.com/librespot-org/librespot/pull/1284 is needed to fix the failing clippy job?

**Review by roderickvd — APPROVED** [review]

**Comment by roderickvd** [comment]
Thanks! Yes the other PR is required to fix the clippy warnings.

**Pull request merged**